eng/pipelines/templates/steps/analyze.yml

```diff
@@ -19,31 +19,44 @@ steps:
     displayName: Dump Package properties
     condition: and(succeeded(), ${{ parameters.IsSdkLibrary }})
 
+  - pwsh: |
+      . $(Build.SourcesDirectory)/eng/scripts/analyze-step-helpers.ps1
+      $directoryToPublish = Join-Path -Path $(Build.ArtifactStagingDirectory) ${{ parameters.ArtifactName }}
+
+      New-GoArtifacts `
+        -ServiceDirectory '${{ parameters.ServiceDirectory }}' `
+        -OutputDirectory $(Build.ArtifactStagingDirectory) `
+        -DirectoryToPublish $directoryToPublish
+
+      Copy-Item "$(Build.ArtifactStagingDirectory)/PackageInfo" -Destination $directoryToPublish -Recurse
+    displayName: 'Create Go Package Artifact'
+
   - template: /eng/common/pipelines/templates/steps/publish-1es-artifact.yml
     parameters:
-      ArtifactName: "PackageInfo"
```

Comment:
Don't we still need old PackageInfo for other EngSys tools?

Reply:
I don't think so. All calls to it are from the $(ArtifactStagingDirectory) where it is initially created.

Reply:
Actually, It is being downloaded in the go release job https://github.com/Azure/azure-sdk-for-go/blob/d6bd6b7ed4e872a5ff0e54bf5bec9fd181339180/eng/pipelines/templates/jobs/archetype-go-release.yml#L63